subsys/bluetooth/host/l2cap_br.c

```diff
@@ -715,6 +715,37 @@ static int l2cap_br_conn_req_reply(struct bt_l2cap_chan *chan, uint16_t result)
 	return 0;
 }
 
+void bt_l2cap_br_chan_del(struct bt_l2cap_chan *chan)
+{
+	const struct bt_l2cap_chan_ops *ops = chan->ops;
+
+	LOG_DBG("conn %p chan %p", chan->conn, chan);
+
+	if (!chan->conn) {
+		goto destroy;
+	}
+
+	if (ops->disconnected) {
+		ops->disconnected(chan);
+	}
+
+	chan->conn = NULL;
+
+destroy:
+#if defined(CONFIG_BT_L2CAP_DYNAMIC_CHANNEL)
```

Comment:
Isn't dynamic channels in practice implied if you have BR/EDR enabled? Even SDP uses dynamic channels (PSM 1). If this isn't already enforced in Kconfig it probably should be.

Reply:
yeah I think we definitely need to touch-up the dynamic-channel compilation scope. 
There are redundant `IS_ENABLED()` and `#ifdef`s. We could probably move everything related to dynamic channels to their own conditionally-compiled files.

Regarding the enforcement, I think @lylezhu2012 can make another (trivial) PR if that's ok.

Reply:
> Regarding the enforcement, I think @lylezhu2012 can make another (trivial) PR if that's ok.

Sure. That's fine with me.